source/math/FloatMatrix.ooc

```diff
@@ -288,6 +301,58 @@ operator * (left: FloatMatrix, right: FloatMatrix) -> FloatMatrix {
 result
 }
 
+// <summary>
+// Addition of matrices.
+// </summary>
+// <param name="left">Left matrix in the addition.</param>
+// <param name="right">Right matrix in the addition.</param>
+// <returns>Sum of left and right matrices.</returns>
+operator + (left: FloatMatrix, right: FloatMatrix) -> FloatMatrix {
+	if (left dimensions width != right dimensions width || left dimensions height != right dimensions height)
+		InvalidDimensionsException new() throw()
+	result := FloatMatrix new (left dimensions width, left dimensions height)
+	for (x in 0 .. right dimensions width) {
+		for (y in 0 .. right dimensions height) {
+			temp := left elements[x + left dimensions width * y] + right elements[x + right dimensions width * y]
+			result elements[x + result dimensions width * y] = temp
+		}
+	}
+result
+}
+
+// <summary>
+// Subtraction of matrices.
+// </summary>
+// <param name="left">Left matrix in the subtraction.</param>
+// <param name="right">Right matrix in the subtraction.</param>
+// <returns>Difference of left and right matrices.</returns>
+operator - (left: FloatMatrix, right: FloatMatrix) -> FloatMatrix {
+	if (left dimensions width != right dimensions width || left dimensions height != right dimensions height)
+		InvalidDimensionsException new() throw()
+	result := FloatMatrix new (left dimensions width, left dimensions height)
+	for (x in 0 .. right dimensions width) {
+		for (y in 0 .. right dimensions height) {
+			temp := left elements[x + left dimensions width * y] - right elements[x + right dimensions width * y]
+			result elements[x + result dimensions width * y] = temp
+		}
+	}
+result
+}
+
+// <summary>
+// Multiplication of scalar and matrix.
+// </summary>
+// <param name="left">Scalar in the multiplication.</param>
+// <param name="right">Matrix in the multiplication.</param>
+// <returns>Product of the sclar and matrix.</returns>
+operator * (left: Float, right: FloatMatrix) -> FloatMatrix {
```

Comment:
All of the operators where the first parameter is a `FloatMatrix` should be moved into the cover. This here operator is the only one I see that needs to be a stand-alone operator (since it takes a `Float` as the first argument).